fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/search/LuceneOptimizedIndexSearcher.java

```diff
@@ -180,8 +205,10 @@ private <C extends Collector> CompletableFuture<C> searchOptimized(@Nonnull fina
                         if (scorer != null && scorer.getLeft() != null && scorer.getRight() != null) {
                             try {
                                 scorer.getRight().score(scorer.getLeft(), ctx.reader().getLiveDocs());
-                            } catch (IOException e) {
+                            } catch (CollectionTerminatedException cte) {
```

Comment:
Based on `IndexSearcher.search`, this can be thrown either in `getLeafCollector` or `score`. So, I think the similar `catch` 14 lines up needs the same treatment.

Reply:
Thanks, I added the same `catch` logic there, and provided some documentation.